lib/src/Uncertainty/StatTests/LinearModelTest.cxx

```diff
@@ -63,70 +121,57 @@ TestResult LinearModelTest::LinearModelResidualMean(const Sample & firstSample,
-  // Fill-in the command file
-  cmdFile << "library(rot)" << std::endl;
-  cmdFile << "options(digits=17)" << std::endl;
-  cmdFile << "options(warn=-1)" << std::endl;
-  cmdFile << "options(stringsAsFactors = F)" << std::endl;
-  cmdFile << "firstSample <- data.matrix(read.table(\"" << firstDataFileName << "\"))" << std::endl;
-  cmdFile << "secondSample <- data.matrix(read.table(\"" << secondDataFileName << "\"))" << std::endl;
-  cmdFile << "regression <- t(data.matrix(read.table(\"" << regressionFileName << "\")))" << std::endl;
-  cmdFile << "res <- test" << testName;
-  cmdFile << "(firstSample, regression, secondSample, " << 1.0 - level << ")" << std::endl;
-  cmdFile << "f <- file(\"" << resultFileName << "\",\"wt\")" << std::endl;
-  cmdFile << "cat(res$test, res$testResult, res$threshold, res$pValue, sep=\"\\n\", file=f)" << std::endl;
-  cmdFile << "close(f)" << std::endl;
-  cmdFile.close();
-  const String RExecutable(ResourceMap::GetAsString("R-executable-command"));
-  OSS systemCommand;
-  if (RExecutable != "") systemCommand << RExecutable << " --no-save --silent < \"" << commandFileName << "\"" << Os::GetDeleteCommandOutput();
-  else throw NotYetImplementedException(HERE) << "In LinearModelTest::RunTwoSamplesALinearModelRTest(const Sample & firstSample, const Sample & secondSample, const LinearModel & linearModel, const Scalar level, const String & testName): needs R. Please install it and set the absolute path of the R executable in ResourceMap.";
-  int returnCode(Os::ExecuteCommand(systemCommand));
-  if (returnCode != 0) throw InternalException(HERE) << "Error: unable to execute the system command " << String(systemCommand) << " returned code is " << returnCode;
-  // Parse result file
-  std::ifstream resultFile(resultFileName.c_str(), std::ios::in);
-  String testType;
-  resultFile >> testType;
-  Bool testResult;
-  resultFile >> testResult;
-  Scalar pThreshold = -1.0;
-  resultFile >> pThreshold;
-  Scalar pValue = -1.0;
-  resultFile >> pValue;
-
-  // Clean-up everything
-  Os::Remove(firstDataFileName);
-  Os::Remove(secondDataFileName);
-  Os::Remove(regressionFileName);
-  Os::Remove(resultFileName);
-  Os::Remove(commandFileName);
-
-  return TestResult(testType, testResult, pValue, pThreshold);
+  const UnsignedInteger dimension = firstSample.getDimension();
+  if (secondSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
+  if (firstSample.getSize() != secondSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
+  if (firstSample.getSize() != secondSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
+  if (firstSample.getSize() < 3) throw InvalidArgumentException(HERE) << "Error: sample too small. Sample should contains at least 3 elements";
+  const UnsignedInteger df = firstSample.getSize() - dimension - 1;
+
```

Comment:
check df. BTW, the same quantity is named df or p or is not named depending on the method. It should be harmonized.